src/test/java/de/zalando/tip/zalenium/proxy/DockerSeleniumRemoteProxyTest.java

```diff
@@ -78,4 +88,54 @@ public void secondRequestGetsANullTestRequest() {
         }
     }
 
+    @Test
+    public void noSessionIsCreatedWhenCapabilitiesAreNotSupported() {
+        // Non supported capabilities
+        Map<String, Object> requestedCapability = new HashMap<>();
+        requestedCapability.put(CapabilityType.BROWSER_NAME, BrowserType.IE);
+        requestedCapability.put(CapabilityType.PLATFORM, Platform.WIN10);
+
+        TestSession newSession = proxy.getNewSession(requestedCapability);
+        Assert.assertNull(newSession);
+    }
+
+    @Test
+    public void pollerThreadTearsDownNodeAfterTestIsCompleted() throws InterruptedException, IOException {
+        // Supported desired capability for the test session
+        Map<String, Object> requestedCapability = new HashMap<>();
+        requestedCapability.put(CapabilityType.BROWSER_NAME, BrowserType.CHROME);
+        requestedCapability.put(CapabilityType.PLATFORM, Platform.LINUX);
+
+        // Start poller thread
+        proxy.startPolling();
+
+        // Mock the poller HttpClient to avoid exceptions due to failed connections
+        HttpClient client = mock(HttpClient.class);
+        HttpResponse httpResponse = mock(HttpResponse.class);
+        StatusLine statusLine = mock(StatusLine.class);
+        when(statusLine.getStatusCode()).thenReturn(200);
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(client.execute(any(HttpPost.class))).thenReturn(httpResponse);
+        proxy.getDockerSeleniumNodePollerThread().setClient(client);
+
+
+        // Get a test session
+        TestSession newSession = proxy.getNewSession(requestedCapability);
+        Assert.assertNotNull(newSession);
+
+        // The node should be busy since there is a session in it
+        Assert.assertTrue(proxy.isBusy());
+
+        // We release the session, the node should be free
+        newSession.getSlot().doFinishRelease();
+
+        // After running one test, the node shouldn't be busy and also down
+        Assert.assertFalse(proxy.isBusy());
+        long sleepTime = proxy.getDockerSeleniumNodePollerThread().getSleepTimeBetweenChecks();
+        await().atMost(sleepTime + 1, MILLISECONDS).untilCall(to(proxy).isDown(), equalTo(true));
+        Thread.sleep(proxy.getDockerSeleniumNodePollerThread().getSleepTimeBetweenChecks() + 1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)